### PR TITLE
Preserve data in the browser

### PR DESCRIPTION
**Pull request by chepkok3**
**Implemented the following interactions:**
-Wrote a JavaScript code that ensures that when the user changes the content of any input field, the data is saved to the local storage.
-When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
**I used the following data model:**
-Created a single JavaScript object with all the data from the entire form and save it in local storage.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -356,4 +356,47 @@ form.addEventListener('submit', (e) => {
     return false;
```
Comment (christianbiring1) [inline script.js:356]: [OPTIONAL]
- [ ] Kindly make sure you use the correct validation for the email field, as you may notice there's no upper case in here but I get this error.
`Hint: You may consider using: `
```js
// the regex expression
/^[^A-Z]+$/
// or comparing the email value to its corresponding lowercase like
email.value == email.value.toLowerCase
// Or switch around your logic in the if-else statement
```
image

File: script.js [hunk 2]
```diff
@@ -356,4 +356,47 @@ form.addEventListener('submit', (e) => {
...
   }
   return form.submit();
+});
+
```
Comment (christianbiring1) [inline script.js:360]: [OPTIONAL]
- [ ] Kindly make sure you stick to the design as much as possible. the spacing, the emplacement of elements must be similar to the design in Figma.

`Hint: You may consider the flexbox to place your element on the page`
Your design :point_down: 
Screenshot from 2023-01-06 16-06-26

The Figma design :point_down: 
image

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @chepkok3 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 

## Highlights
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 
- [ ] The user inputs can be saved in the localStorage :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 2 above)
Comment (chepkok3) [reply script.js:360]: @christianbiring1 I am not getting your review clearly. Are you supposed to review the preserve data in the browser pull request or the contact form validation which has already been approved and I have noted the changes that I will make?

Comment (chepkok3) [reply script.js:360]: @christianbiring1 What is your review for the preserve data in the browser pull request and why are all these change requests optional but you have blocked the project from being completed?

File: script.js (hunk 1 above)
Comment (Whoistolu) [reply script.js:356]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @chepkok3 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 2 above)
Comment (christianbiring1) [reply script.js:360]: - @chepkok3 I couldn't approve the project and some test cases for the previous milestone were not working correctly.
Apart from your project being approved by the previous reviewer maybe while implementing the new changes you broke the code and that's why that misfunctioning
I couldn't approve a project that is not presentable

**Pull request merged**